### PR TITLE
added code to getVariantCalls to cope with "no variants" in requested region

### PR DESCRIPTION
**Pull request by vjcitn**
A new parameter (nullaction) was introduced to control the action upon requesting
a region that has no variants -- can return NULL or stop() depending on
setting of nullaction

Also improved name handling for unnamed variants, testing against additional variant datasets in GCP

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Package: GoogleGenomics
-Version: 1.5.4
+Version: 1.5.6
```
Comment (siddharthab) [inline DESCRIPTION]: Please change version to 1.7.3.

File: R/variants.R [hunk 2]
```diff
@@ -248,7 +249,7 @@ variantsToGRanges <- function(variants, oneBasedCoord=TRUE, slStyle="UCSC") {
 getVariantCalls = function(datasetId = "10473108253681171589", 
     chromosome = "22", 
     start = 16051400, end = 16051500, fields = NULL, converter = c,
-    oneBasedCoord = TRUE)  {
+    oneBasedCoord = TRUE, nullaction="stop")  {
```
Comment (siddharthab) [inline R/variants.R]: Please change to nullAction everywhere and add a space around =.

`nullAction = "stop"`

File: R/variants.R [hunk 3]
```diff
@@ -271,6 +272,13 @@ ggv = getVariants(datasetId = datasetId, chromosome=chromosome,
 # obtain start, end, chr, and base for each variant in requested range
 #
 sts = as.numeric(sapply(ggv, function(x) x$start))
+if (length(sts)==0) {
```
Comment (siddharthab) [inline R/variants.R]: Correctly indented block for better readability:

```
if (length(sts) == 0) {
  if (nullAction == "warn") {
    warning("No variants in this interval.")
    return(NULL)
  }
  else stop("No variants in this interval.")
}
```

File: R/variants.R [hunk 4]
```diff
@@ -279,7 +287,11 @@ alts = sapply(ggv, function(x) x$alternateBases)
 # variant names could be one per alt
 #
 vnames = sapply(ggv, "[[", "names")
-allnames = apply(vnames,2,function(x) paste(unique(x), collapse=";"))
+if (is.matrix(vnames))
```
Comment (siddharthab) [inline R/variants.R]: Spaces after ',' and curly braces around second lambda function.

```
if (is.matrix(vnames))
  allnames = apply(vnames, 2, function(x) paste(unique(x), collapse=";"))
else if (is.list(vnames))
  allnames = sapply(vnames, function(x)
      {if (is.character(unlist(x))) paste(unique(x), collapse=";") else NA})
```

**Comment by shwetagopaul92** [comment]
@siddharthab Required changes have been made and committed. Thank you!

**Comment by siddharthab** [comment]
Thank you for your contribution. I will push the changes to bioc-devel and notify here.

**Comment by siddharthab** [comment]
Sorry for the delay; these changes should now be available in bioc-devel repository.

**Pull request merged**